source/OptimaJet.DWKit.StarterApplication/Services/HttpCurrentUserContext.cs

```diff
@@ -1,20 +1,62 @@
 ﻿using System;
+using Auth0.ManagementApi;
+using Auth0.ManagementApi.Models;
 using Microsoft.AspNetCore.Http;
-using Optimajet.DWKit.StarterApplication.Models;
 using OptimaJet.DWKit.StarterApplication.Utility;
+using static OptimaJet.DWKit.StarterApplication.Utility.EnvironmentHelpers;
+using Serilog;
 
 namespace OptimaJet.DWKit.StarterApplication.Services
 {
     public class HttpCurrentUserContext : ICurrentUserContext
     {
         public HttpContext HttpContext { get; set; }
+        public Auth0ManagementApiTokenService TokenService { get; set; }
 
         private string auth0Id;
+        private string email;
+        private string givenName;
+        private string familyName;
+        private string name;
+        private string authType;
+        private ManagementApiClient managementApiClient;
+        private User auth0User;
 
         public HttpCurrentUserContext(
-            IHttpContextAccessor httpContextAccessor)
+            IHttpContextAccessor httpContextAccessor,
+            Auth0ManagementApiTokenService tokenService)
         {
             this.HttpContext = httpContextAccessor.HttpContext;
+            this.TokenService = tokenService;
+        }
+
+        private string AuthType {
+            get {
+                if (authType == null) {
+                    authType = this.HttpContext.GetAuth0Type();
+                }
+                return authType;
+            }
+        }
+
+        private ManagementApiClient ManagementApiClient {
+            get {
+                if (managementApiClient == null) {
+                    var token = TokenService.Token;
+                    var domainUri = new Uri(GetVarOrThrow("AUTH0_DOMAIN"));
+                    managementApiClient = new ManagementApiClient(token, domainUri.Host);
+                }
+                return managementApiClient;
+            }
+        }
+
+        private User Auth0User {
```

Comment:
not sure if you want to denote the difference between an Auth0 User and our users here, but it may be good to point out? idk

Reply:
It is a private property and the using statement is for Auth0 User.  There are other User objects in other namespaces.  Other option is to not use using.

Reply:
whatever is fine by me